drivers/mipi_dsi/dsi_stm32.c

```diff
@@ -457,6 +507,12 @@ static int mipi_dsi_stm32_init(const struct device *dev)
 			.enr = DT_INST_CLOCKS_CELL_BY_NAME(inst, dsiclk, bits),			\
 			.bus = DT_INST_CLOCKS_CELL_BY_NAME(inst, dsiclk, bus),			\
 		},										\
+		COND_CODE_1(DT_INST_CLOCKS_HAS_NAME(inst, dsisrc),				\
+			(.dsisrc_clk = {							\
+				.enr = DT_INST_CLOCKS_CELL_BY_NAME(inst, dsisrc, bits),		\
+				.bus = DT_INST_CLOCKS_CELL_BY_NAME(inst, dsisrc, bus),		\
+			},),									\
+			(.dsisrc_clk = {0},))							\
```

Comment:
Non blocking, for readability and better fit with other clocks
```suggestion
		.dsisrc_clk =  COND_CODE_1(DT_INST_CLOCKS_HAS_NAME(inst, dsisrc),		\
				({							\
				.enr = DT_INST_CLOCKS_CELL_BY_NAME(inst, dsisrc, bits),		\
				.bus = DT_INST_CLOCKS_CELL_BY_NAME(inst, dsisrc, bus),		\
			},),{0})
```